modular_skyrat/modules/xenos_skyrat_redo/code/xeno_types/warrior.dm

```diff
@@ -0,0 +1,86 @@
+/// SKYRAT MODULE SKYRAT_XENO_REDO
+
+/mob/living/carbon/alien/humanoid/skyrat/warrior
+	name = "alien warrior"
+	desc = "If there are aliens to call walking tanks, this would be one of them, with both the heavy armor and strong arms to back that claim up."
+	caste = "warrior"
+	maxHealth = 400
+	health = 400
+	icon_state = "alienwarrior"
+	melee_damage_lower = 30
+	melee_damage_upper = 35
+	/// Holds the charge ability that will be given to the warrior later
+	var/datum/action/cooldown/mob_cooldown/charge/basic_charge/defender/charge
+	/// Holds the tail sweep ability that will be given to the warrior later
+	var/datum/action/cooldown/spell/aoe/repulse/xeno/skyrat_tailsweep/tail_sweep
+	/// Holds the agility ability that will be given to the warrior later
+	var/datum/action/cooldown/alien/skyrat/warrior_agility/agility
+
+/mob/living/carbon/alien/humanoid/skyrat/warrior/Initialize(mapload)
+	. = ..()
+	tail_sweep = new /datum/action/cooldown/spell/aoe/repulse/xeno/skyrat_tailsweep()
+	tail_sweep.Grant(src)
+
+	charge = new /datum/action/cooldown/mob_cooldown/charge/basic_charge/defender()
+	charge.Grant(src)
+
+	agility = new /datum/action/cooldown/alien/skyrat/warrior_agility()
+	agility.Grant(src)
+
+	REMOVE_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+	add_movespeed_modifier(/datum/movespeed_modifier/alien_big)
+
+/mob/living/carbon/alien/humanoid/skyrat/warrior/Destroy()
+	QDEL_NULL(charge)
+	QDEL_NULL(tail_sweep)
+	QDEL_NULL(agility)
+	return ..()
+
+/mob/living/carbon/alien/humanoid/skyrat/warrior/create_internal_organs()
+	internal_organs += new /obj/item/organ/internal/alien/plasmavessel
+	..()
+
+/datum/action/cooldown/alien/skyrat/warrior_agility
+	name = "Agility Mode"
+	desc = "Drop onto all fours, increasing your speed at the cost of being unable to use most abilities."
```

Comment:
this should probably drop their attack damage a bit, it's not as bad as everything else, but 30-35 with a pretty-fast movespeed is a bit much